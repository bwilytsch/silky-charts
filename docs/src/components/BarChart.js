import React from 'react';
import styled from 'styled-components';
import { Bar } from 'silky-charts';
import Chart from '../containers/Chart';
import BarData from '../markdown/bar-data.md';
import BarProps from '../markdown/bar-props.md';
import { InlineCode, Pre } from '../ui/code';
import { Table, Td } from '../ui/table';

const data = [
  { name: 'E', value: 0.12702 },
  { name: 'T', value: 0.09056 },
  { name: 'A', value: 0.08167 },
  { name: 'O', value: 0.07507 },
  { name: 'I', value: 0.06966 },
  { name: 'N', value: 0.06749 },
  { name: 'S', value: 0.06327 },
  { name: 'H', value: 0.06094 },
  { name: 'R', value: 0.05987 },
  { name: 'D', value: 0.04253 },
  { name: 'L', value: 0.04025 },
  { name: 'C', value: 0.02782 },
  { name: 'U', value: 0.02758 },
  { name: 'M', value: 0.02406 },
  { name: 'W', value: 0.0236 },
  { name: 'F', value: 0.02288 },
  { name: 'G', value: 0.02015 },
  { name: 'Y', value: 0.01974 },
  { name: 'P', value: 0.01929 },
  { name: 'B', value: 0.01492 },
  { name: 'V', value: 0.00978 },
  { name: 'K', value: 0.00772 },
  { name: 'J', value: 0.00153 },
  { name: 'X', value: 0.0015 },
  { name: 'Q', value: 0.00095 },
  { name: 'Z', value: 0.00074 },
];

const H3 = styled.h3`
  margin-top: 40px;
`;

const BarChart = () => (
  <>
    <h2>Bar Chart</h2>
    <Chart>
      <Bar data={data} responsive tooltip />
    </Chart>
    <H3>Data Definition</H3>
    <BarData components={{ pre: Pre }} />
    <H3>Props</H3>
    <BarProps components={{ inlineCode: InlineCode, table: Table, td: Td }} />
  </>
);

export default BarChart;
