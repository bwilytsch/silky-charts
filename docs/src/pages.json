[
  {
    "title": "Getting Started",
    "path": "/",
    "component": "Introduction"
  },
  {
    "title": "Bar",
    "path": "bar",
    "component": "BarChart"
  }
]
