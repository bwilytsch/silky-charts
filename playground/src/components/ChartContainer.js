import styled from 'styled-components';

const ChartContainer = styled.div`
  background-color: rgb(255, 255, 255);
  border-radius: 10px;
  width: 80%;
`;

export default ChartContainer;
