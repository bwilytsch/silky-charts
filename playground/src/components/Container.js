import styled from 'styled-components';

const Container = styled.div`
  align-items: center;
  background-color: rgba(0, 122, 255, 0.1);
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
`;

export default Container;
