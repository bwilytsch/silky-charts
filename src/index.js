export { default as Bar } from './Bar';
export { default as BarLine } from './BarLine';
export { default as StackedArea } from './StackedArea';

export { default as toSeries } from './utils/toSeries';
