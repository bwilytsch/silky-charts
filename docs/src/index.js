import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const baseURL = '/silky-charts';
  const pathname = window.location.pathname;
  const currentPath = pathname.includes(baseURL)
    ? pathname.replace(baseURL, '')
    : '/';
  const isValidPathname =
    currentPath && pages.map(({ path }) => path).includes(currentPath);

  console.log(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        baseURL={baseURL}
        initial={isValidPathname ? currentPath : '/'}
      />
      <Router primary={false}>
        {currentPath === '/' && (
          <Redirect from={currentPath} to={baseURL} noThrow />
        )}
        <Content path={baseURL}>
          {pages.map(({ component, path }, idx) => {
            const Component = contentComponents[component];
            return <Component key={idx} path={path} default={path === '/'} />;
          })}
        </Content>
      </Router>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
