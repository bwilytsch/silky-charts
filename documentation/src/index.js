import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const pathname = window.location.pathname;
  const defaultPath = '/';
  const currentPath =
    pathname !== defaultPath ? pathname.replace(defaultPath, '') : pathname;
  const isInvalidPath = !pages
    .map(({ pathname }) => pathname)
    .includes(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        initialPath={isInvalidPath ? defaultPath : currentPath}
      />
      <Content>
        <Router>
          {isInvalidPath && (
            <Redirect from={currentPath} to={defaultPath} noThrow />
          )}
          {pages.map(({ component, pathname }, idx) => {
            const Component = contentComponents[component];
            return (
              <Component
                key={idx}
                path={pathname}
                default={pathname === defaultPath}
              />
            );
          })}
        </Router>
      </Content>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
