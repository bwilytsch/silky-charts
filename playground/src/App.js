/* eslint-disable */

import React from 'react';
import styled from 'styled-components';
import { Bar, BarLine, StackedArea, toSeries } from 'silky-charts';
import { aapl, fruits, letters, transport } from './data/';
import Container from './components/Container';
import ChartContainer from './components/ChartContainer';
import 'normalize.css';
import './styles.css';

const App = () => (
  <Container>
    <ChartContainer>
      <StackedArea
        // aspectRatio="4:3"
        data={fruits}
        // dataSource="Unknown"
        // dateFormat="%B %d, %Y"
        // grid
        // height="300"
        // lineSeries={['Sold', 'Lost']}
        // lineType="curveCardinalOpen"
        // lineTypeOption={0.1}
        margin={{ top: 100, right: 100, bottom: 100, left: 100 }}
        // onClick={() => {}}
        // onMouseEnter={() => {}}
        // onMouseLeave={() => {}}
        // padding={1}
        responsive
        // secondaryTheme="madang"
        // stackedSeries={['Pear', 'Mango', 'Papaya']}
        // staticTooltip="top-left"
        // theme="curiousBlue"
        // title="Placeholder Chart Title"
        tooltip
        // width={500}
        // xAxisChartLabel="X axis label"
        // xAxisLabelRotation
        // xAxisLabelRotationValue={-65}
        // xAxisTicks={10}
        // yAxisChartLabel="Y axis label"
        // yAxisTicks={10}
      />
    </ChartContainer>
  </Container>
);

export default App;
