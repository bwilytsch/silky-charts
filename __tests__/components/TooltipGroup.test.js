import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { TooltipGroup } from '../../src/components';
import 'jest-styled-components';

afterEach(cleanup);

test('TooltipGroup', () => {
  const { container } = render(
    <TooltipGroup
      data={[
        { name: 'foo', series: 'bar', value: 0 },
        { name: 'foo', series: 'qux', value: 1 },
      ]}
    />
  );

  expect(container).toMatchSnapshot();
  expect(container.firstChild.textContent).toEqual('foo');
  expect(container.childNodes.length).toEqual(3);
  expect(container.childNodes[1].textContent).toEqual('0 on bar');
  expect(container.childNodes[2].textContent).toEqual('1 on qux');
});

test('TooltipGroup with date', () => {
  const date = new Date('09/24/1980');
  const { container } = render(
    <TooltipGroup
      data={[
        { name: date, series: 'bar', value: 0 },
        { name: date, series: 'qux', value: 1 },
      ]}
    />
  );

  expect(container).toMatchSnapshot();
  expect(container.firstChild.textContent).toEqual('Sep 24, 1980');
  expect(container.childNodes.length).toEqual(3);
  expect(container.childNodes[1].textContent).toEqual('0 on bar');
  expect(container.childNodes[2].textContent).toEqual('1 on qux');
});
