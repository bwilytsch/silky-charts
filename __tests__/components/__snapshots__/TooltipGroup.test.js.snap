// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipGroup 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1:not(:last-child) {
  margin-bottom: 6px;
}

.c2 {
  background-color: rgb(8,104,172);
  margin-right: 2px;
  padding: 0 3px;
}

.c3 {
  background-color: rgb(8,104,172);
  margin-right: 10px;
  padding: 0 1px;
}

.c6 {
  background-color: rgb(67,162,202);
  margin-right: 2px;
  padding: 0 3px;
}

.c7 {
  background-color: rgb(67,162,202);
  margin-right: 10px;
  padding: 0 1px;
}

.c0 {
  color: grey;
  font-size: 12px;
  font-weight: 600;
  margin-bottom: 10px;
  text-align: right;
}

.c4 {
  color: rgb(255,255,255);
  font-size: 12px;
  font-weight: 600;
  margin-bottom: 2px;
}

.c5 {
  color: grey;
  font-weight: 500;
}

<div>
  <div
    class="c0"
  >
    foo
  </div>
  <div
    class="c1"
  >
    <span
      class="c2"
      width="6"
    />
    <span
      class="c3"
      width="2"
    />
    <div
      class="c4"
    >
      0
       
      <span
        class="c5"
      >
        on
      </span>
       
      bar
    </div>
  </div>
  <div
    class="c1"
  >
    <span
      class="c6"
      width="6"
    />
    <span
      class="c7"
      width="2"
    />
    <div
      class="c4"
    >
      1
       
      <span
        class="c5"
      >
        on
      </span>
       
      qux
    </div>
  </div>
</div>
`;

exports[`TooltipGroup with date 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1:not(:last-child) {
  margin-bottom: 6px;
}

.c2 {
  background-color: rgb(8,104,172);
  margin-right: 2px;
  padding: 0 3px;
}

.c3 {
  background-color: rgb(8,104,172);
  margin-right: 10px;
  padding: 0 1px;
}

.c6 {
  background-color: rgb(67,162,202);
  margin-right: 2px;
  padding: 0 3px;
}

.c7 {
  background-color: rgb(67,162,202);
  margin-right: 10px;
  padding: 0 1px;
}

.c0 {
  color: grey;
  font-size: 12px;
  font-weight: 600;
  margin-bottom: 10px;
  text-align: right;
}

.c4 {
  color: rgb(255,255,255);
  font-size: 12px;
  font-weight: 600;
  margin-bottom: 2px;
}

.c5 {
  color: grey;
  font-weight: 500;
}

<div>
  <div
    class="c0"
  >
    Sep 24, 1980
  </div>
  <div
    class="c1"
  >
    <span
      class="c2"
      width="6"
    />
    <span
      class="c3"
      width="2"
    />
    <div
      class="c4"
    >
      0
       
      <span
        class="c5"
      >
        on
      </span>
       
      bar
    </div>
  </div>
  <div
    class="c1"
  >
    <span
      class="c6"
      width="6"
    />
    <span
      class="c7"
      width="2"
    />
    <div
      class="c4"
    >
      1
       
      <span
        class="c5"
      >
        on
      </span>
       
      qux
    </div>
  </div>
</div>
`;
